core/cluster/process.go

```diff
@@ -0,0 +1,38 @@
+package cluster
+
+import "time"
+
+type (
+
+	SingleProcess interface {
+		//Do runs the given function if there is no other function with that key running
+		Do(key string, fn func()) (bool, error)
+	}
+
+	DistributedLock interface {
+		//GetLock - call to requests a distributed lock - if not possible. returns "false" in case Lock is not available without blocking
+		GetLock(key string, max *time.Duration) (bool, error)
+		//Free the distributed Lock
+		Unlock(key string)
```

Comment:
Can `Unlock` fail? As we do rely on network operation for distribution handling we might want to keep that into account?